File: dashboards/cupid/frontend/src/api/sessions.ts
import { fetchApi } from './client'
import type { Session, SessionStats, ConversationResponse, TimeRange } from '@/types'

interface SessionsResponse {
  data: Session[]
  meta: {
    total: number
    page: number
    limit: number
    pages: number
  }
}

export async function getSessions(params: {
  time_range?: TimeRange
  status?: 'complete' | 'incomplete' | 'all'
  search?: string
  page?: number
  limit?: number
}): Promise<SessionsResponse> {
  const searchParams = new URLSearchParams()
  if (params.time_range) searchParams.set('time_range', params.time_range)
  if (params.status) searchParams.set('status', params.status)
  if (params.search) searchParams.set('search', params.search)
  if (params.page) searchParams.set('page', params.page.toString())
  if (params.limit) searchParams.set('limit', params.limit.toString())

  return fetchApi(`/sessions?${searchParams}`)
}

export async function getSessionStats(time_range?: TimeRange): Promise<SessionStats> {
  const searchParams = new URLSearchParams()
  if (time_range) searchParams.set('time_range', time_range)
  return fetchApi(`/sessions/stats?${searchParams}`)
}

export async function getConversation(sessionId: string): Promise<ConversationResponse> {
  return fetchApi(`/sessions/${sessionId}/conversation`)
}

File: dashboards/cupid/backend/app/routers/sessions.py
from fastapi import APIRouter, HTTPException, Query
from typing import Literal

from app.services import AnalyticsService

router = APIRouter(prefix="/api/sessions", tags=["sessions"])
analytics = AnalyticsService()


@router.get("")
async def list_sessions(
    time_range: Literal["24h", "7d", "30d", "all"] = "all",
    status: Literal["complete", "incomplete", "all"] = "all",
    search: str | None = None,
    page: int = Query(1, ge=1),
    limit: int = Query(50, ge=1, le=100),
):
    """List sessions with filters."""
    return await analytics.get_sessions(
        time_range=time_range,
        status=status,
        search=search,
        page=page,
        limit=limit,
    )


@router.get("/stats")
async def get_session_stats(
    time_range: Literal["24h", "7d", "30d", "all"] = "all",
):
    """Get aggregate session statistics."""
    return await analytics.get_session_stats(time_range=time_range)


@router.get("/{session_id}/conversation")
async def get_conversation(session_id: str):
    """Get full conversation for a session."""
    result = await analytics.get_conversation(session_id)
    if not result.get("session"):
        raise HTTPException(status_code=404, detail="Session not found")
    return result
